x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -829,6 +831,30 @@ private static void checkJoin(LogicalPlan plan, Set<Failure> failures) {
                 }
             }
 
+            if (join instanceof LookupJoin) {
+                join.right().forEachDown(EsRelation.class, relation -> {
+                    var indicesWithModes = relation.index().indexNameWithModes();
```

Comment:
Suggestion (if it's not too much work):

Instead of doing this in the verifier, we can move this directly into `Analyzer.resolveTable`. We can return an `UnresolvedRelation` with a custom message.

That may seem a bit arbitrary, but I think it will make our lives easier later; the reason is that we shouldn't slap the whole `EsIndex` object on the `EsRelation` because it's heavy; I think we want to remove that from `EsRelation` which  will make our serialized plans smaller: https://github.com/elastic/elasticsearch/issues/112998

Reply:
Moved it, looks good